api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/oauth/OAuthProperties.java

```diff
@@ -36,6 +37,14 @@
      */
     private boolean replicateSessions;
 
+    /**
+     * The pac4j framework sets a CSRF cookie as part of its Oauth configuration.
+     * This allows the pac4j cookie property defaults to be overridden but the cookie name, pinToSession, and comment
+     * do are not used by pac4j. Defaults are set to pac4j defaults.
+     */
+    @NestedConfigurationProperty
+    private CookieProperties csrfCookie = new CsrfCookieProperties();
```

Comment:
To help with metadata generation, use `CsrfCookieProperties` as the defining type.